src/agent/application.cpp

```diff
@@ -255,7 +255,11 @@ void Application::InitRcpMode(void)
     mPublisher->Start();
 #endif
 #if OTBR_ENABLE_BORDER_AGENT
-    mBorderAgent->Init();
+    mHost.BorderAgentSetMeshCoPServiceChangedCallback(
+        [this](bool aIsActive, uint16_t aPort, const uint8_t *aTxtData, uint16_t aLength) {
+            mBorderAgent->HandleBorderAgentStateChange(aIsActive, aPort);
+            mBorderAgent->HandleOtMeshCoPTxtValueChange(std::vector<uint8_t>(aTxtData, aTxtData + aLength));
```

Comment:
Seems like there's a chance that it'll publish mesh cop services twice in a row? Can we merge the updates?

Reply:
Updated.